packages/in_app_purchase/in_app_purchase/README.md

```diff
@@ -109,53 +110,72 @@ class _MyAppState extends State<MyApp> {
     if (purchaseDetails.status == PurchaseStatus.pending) {
-      _showPendingUI();
+      showPendingUI();
     } else {
       if (purchaseDetails.status == PurchaseStatus.error) {
-        _handleError(purchaseDetails.error!);
+        handleError(purchaseDetails.error!);
       } else if (purchaseDetails.status == PurchaseStatus.purchased ||
-                 purchaseDetails.status == PurchaseStatus.restored) {
-        bool valid = await _verifyPurchase(purchaseDetails);
+          purchaseDetails.status == PurchaseStatus.restored) {
+        final bool valid = await _verifyPurchase(purchaseDetails);
         if (valid) {
-          _deliverProduct(purchaseDetails);
+          unawaited(deliverProduct(purchaseDetails));
         } else {
           _handleInvalidPurchase(purchaseDetails);
+          return;
+        }
+      }
+      if (Platform.isAndroid) {
+        if (!_kAutoConsume && purchaseDetails.productID == _kConsumableId) {
+          final InAppPurchaseAndroidPlatformAddition androidAddition =
+              _inAppPurchase.getPlatformAddition<
+                  InAppPurchaseAndroidPlatformAddition>();
+          await androidAddition.consumePurchase(purchaseDetails);
         }
       }
       if (purchaseDetails.pendingCompletePurchase) {
-        await InAppPurchase.instance
-            .completePurchase(purchaseDetails);
+        await _inAppPurchase.completePurchase(purchaseDetails);
       }
     }
-  });
+  }
 }
 ```
 
 ### Connecting to the underlying store
 
+<?code-excerpt "example/lib/main.dart (ConnectToStore)" plaster="none"?>
 ```dart
-final bool available = await InAppPurchase.instance.isAvailable();
-if (!available) {
+final bool isAvailable = await _inAppPurchase.isAvailable();
+if (!isAvailable) {
   // The store cannot be reached or accessed. Update the UI accordingly.
 }
 ```
 
 ### Loading products for sale
 
+<?code-excerpt "example/lib/main.dart (LoadProducts)" plaster="none"?>
 ```dart
-// Set literals require Dart 2.2. Alternatively, use
-// `Set<String> _kIds = <String>['product1', 'product2'].toSet()`.
-const Set<String> _kIds = <String>{'product1', 'product2'};
-final ProductDetailsResponse response =
-    await InAppPurchase.instance.queryProductDetails(_kIds);
-if (response.notFoundIDs.isNotEmpty) {
-  // Handle the error.
-}
-List<ProductDetails> products = response.productDetails;
+const String _kConsumableId = 'consumable';
+const String _kUpgradeId = 'upgrade';
+const String _kSilverSubscriptionId = 'subscription_silver';
+const String _kGoldSubscriptionId = 'subscription_gold';
+const List<String> _kProductIds = <String>[
+  _kConsumableId,
+  _kUpgradeId,
+  _kSilverSubscriptionId,
+  _kGoldSubscriptionId,
+];
+    final ProductDetailsResponse productDetailResponse =
+        await _inAppPurchase.queryProductDetails(_kProductIds.toSet());
+    if (productDetailResponse.error != null) {
+      // Handle the error.
+    }
+      _products = productDetailResponse.productDetails;
 ```
```

Comment:
I find the new example code more verbose and harder to follow than the old code (e.g., not having an inline type for `products`). This seems worth using a separate excerpts file.

Reply:
What do you think about the discrepancy between how the README code was checking for errors (with `response.notFoundIDs.isNotEmpty`) and how the example app did it (with `response.error != null` and `response.productDetails.isEmpty`)? It seems like all three might be useful to include in the README?